File: shared/contracts/typescript/claimSummary.ts
// ClaimSummary contract
// Represents the governed AI-generated summary of a claim.

import type { EvidenceSource } from "./evidenceSource";
import type { GovernanceDecision } from "./governanceDecision";
import type { ModelRouteDecision } from "./modelRouteDecision";

export interface ClaimSummary {
  summaryId: string;
  claimId: string;
  generatedAt: string;         // ISO 8601

  // Core AI output
  summary: string;
  keyFindings: string[];       // Bullet-point findings extracted by the model
  coverageAnalysis: string;    // Which coverages apply and why
  recommendedActions: string[]; // Next steps the adjuster should consider

  // Grounding and provenance
  evidenceSources: EvidenceSource[];

  // Governance and routing metadata
  governanceDecision: GovernanceDecision;
  modelRouteDecision: ModelRouteDecision;

  // Confidence
  confidenceScore: number;     // 0.0 – 1.0
  confidenceRationale: string;
}

File: shared/contracts/python/claim_summary.py
"""
ClaimSummary contract — Python / Pydantic.
Represents the governed AI-generated summary of a claim.
"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

from .evidence_source import EvidenceSource
from .governance_decision import GovernanceDecision
from .model_route_decision import ModelRouteDecision


class ClaimSummary(BaseModel):
    summary_id: str
    claim_id: str
    generated_at: datetime

    # Core AI output
    summary: str
    key_findings: list[str] = Field(default_factory=list)
    coverage_analysis: str
    recommended_actions: list[str] = Field(default_factory=list)

    # Grounding and provenance
    evidence_sources: list[EvidenceSource] = Field(default_factory=list)

    # Governance and routing metadata
    governance_decision: GovernanceDecision
    model_route_decision: ModelRouteDecision

    # Confidence
    confidence_score: float = Field(ge=0.0, le=1.0)
    confidence_rationale: str
